storage/google/cloud/storage/client.py

```diff
@@ -60,6 +62,22 @@ def __init__(self, project=None, credentials=None, _http=None):
         self._connection = Connection(self)
         self._batch_stack = _LocalStack()
 
+    @classmethod
+    def create_anonymous_client(cls):
+        """Factory: return client w/ anonymous credentials.
+        
+        .. note::
+
+           Such a client has only limited access to "public" buckets:
+           listing their contents and downloading their blobs.
+           
+        :rtype: :class:`google.cloud.storage.client.Client`
+        :returns: Instance w/ anonymous credentials and no project.
+        """
+        client = cls(project='<none>', credentials=AnonymousCredentials())
```

Comment:
@tseaver Is there any reason not to also define `Client.ANONYMOUS = AnonymousCredentials()` so users can re-use?

Reply:
I figured YAGNI:  originally, I had planned to have the user pass in an `AnonymousCredentials` instance directly, but this method makes that path unimportant.  It also provides a place to document the expected usage pattern / restrictions.

Reply:
Sphinx allows documentation on `attr`s, so we could also document expected usage pattern / restrictions on the class constant:

```python
class Client(...):
    ANONYMOUS = AnonymousCredentials()
    """Unauthenticated credentials to be used for public buckets.

    For example:

    .. code-block:: python

       >>> import foo
    ...
    """
```

Reply:
Seems like unnecessary indirection to me.

Reply:
It may be useful, especially if we want to check for identity:

```python
if self._credentials is Client.ANONYMOUS:
    ...
```

Though I suppose an `isinstance()` check would be fine too.

Reply:
Yeah it's not a singleton, so I would never want us to rely on a specific anonymouscredentials instance.